chacra/controllers/binaries/__init__.py

```diff
@@ -128,6 +128,35 @@ def index_put(self):
         self.binary.path = self.save_file(file_obj)
         return dict()
 
+    @secure(basic_auth)
+    @index.when(method='DELETE', template='json')
+    def index_delete(self):
+        if not self.binary:
+            abort(404)
+        binary_path = self.binary.path
+        repo = self.binary.repo
+        project = self.binary.project
+        delete(self.binary)
+        try:
+            os.remove(binary_path)
+        except IOError:
+            # should we complain loudly here?
+            logger.exception("Could not remove the binary path: %s" % binary_path)
+        commit()
```

Comment:
My thought was that I needed the commit here so that the query below that's checking binaries for a repo or project would not include the binary I just deleted, but maybe that's not the case.

I'll test that.